src/types.rs

```diff
@@ -454,19 +430,19 @@ impl Default for RuntimeConfig {
 			block_matrix_partition: None,
 			sync_blocks_depth: None,
 			max_cells_per_rpc: Some(30),
-			kad_record_ttl: defaults::ttl(),
-			threshold: defaults::threshold(),
-			replication_factor: defaults::replication_factor(),
-			publication_interval: defaults::publication_interval(),
-			replication_interval: defaults::replication_interval(),
-			connection_idle_timeout: defaults::connection_idle_timeout(),
-			query_timeout: defaults::query_timeout(),
-			query_parallelism: defaults::query_parallelism(),
-			caching_max_peers: defaults::caching_max_peers(),
+			kad_record_ttl: 24 * 60 * 60,
+			threshold: 5000,
+			replication_factor: 20,
+			publication_interval: 12 * 60 * 60,
+			replication_interval: 3 * 60 * 60,
+			connection_idle_timeout: 30,
+			query_timeout: 60,
+			query_parallelism: 3,
+			caching_max_peers: 1,
 			disjoint_query_paths: false,
-			max_kad_record_number: defaults::max_kad_record_number(),
-			max_kad_record_size: defaults::max_kad_record_size(),
-			max_kad_provided_keys: defaults::max_kad_provided_keys(),
+			max_kad_record_number: 2400000,
+			max_kad_record_size: 100,
+			max_kad_provided_keys: 1024,
```

Comment:
Marking specific field as default and providing default function enables omitting that field in config file (and reusing it in default struct value). Does this code preserve that behavior? And if You know, how is doing that? I guess default struct value needs to be instantiated to use some value from it (in case when one field is missing for example).

Reply:
@luka-ethernal showed me docs that explains that behavior. Additional considerations could be how to enforce mandatory values, and what to do in case of optionals (e.g. if field is omitted, user could expect None instead of default value which could be different).

Reply:
> Additional considerations could be how to enforce mandatory values, and what to do in case of optionals (e.g. if field is omitted, user could expect None instead of default value which could be different).
Agreed. 
For optionals, does `confy` support them? Could not find anything in the doc which suggests so. It requires every field to have Default implemented. 
For the defaults, I have tried to capture all of them in the README so that the second confusion could be avoided as much as possible.

Reply:
Lol I was able to post second comment but wasnt able to see yours reply before refreshing page, thanks github. Makes sense, we will just briefly try to cover mandatory and optional cases.

Reply:
After discussion it seems that mandatory field case can be avoided which will reduce complexity and that optional cases can be handled separately so Ill resolve my comment.